packages/codegen-sdk/src/parse-operation.ts

```diff
@@ -84,6 +85,22 @@ export function parseOperations(
     const model = models.find(b => b.name === fragment?.name.value);
     const modelName = model?.name ?? "UNKNOWN_MODEL";
 
+    /** Find a parent operation */
+    const parentSdkKey = sdkPath.slice(0, -1).join("_");
+    const parent = acc[parentSdkKey]?.operations.find(operation => operation.path.join("_") === sdkKey);
+
+    /** Build the path through the response to the data */
+    const responsePath = parent
+      ? printList([parent?.print.responsePath, returnedField?.name?.value], parent?.nonNull ? `.` : "?.")
+      : printList([Sdk.RESPONSE_NAME, returnedField?.name?.value], ".");
+
+    /** Identify whether the operation response data is non-nullable */
+    const nonNullField = reduceNonNullType(
+      parent?.model?.fields.all.find(f => f.name === returnedField?.name?.value)?.node.type ?? ""
+    );
+    const nonNullQuery = query?.type && reduceNonNullType(query?.type);
+    const nonNull = Boolean(nonNullQuery || ((parent ? parent?.nonNull : true) && nonNullField));
```

Comment:
We need to ensure either the operation is a query, that returns a non nullable. Or that the field from the parent is non nullable and the parent is also non nullable